### PR TITLE
refactor: remove deprecated save option for anonymous_id_for_user

### PR DESCRIPTION
**Pull request by dyudyunov**
https://github.com/openedx/public-engineering/issues/35

YT: https://youtrack.raccoongang.com/issue/OeX_Depr-79

### CODE REVIEW

File: common/djangoapps/student/models.py [hunk 1]
```diff
@@ -180,13 +178,6 @@ def anonymous_id_for_user(user, course_id, save='DEPRECATED'):
     # This part is for ability to get xblock instance in xblock_noauth handlers, where user is unauthenticated.
     assert user
 
-    if save != 'DEPRECATED':
```
Comment (idegtiarov) [inline common/djangoapps/student/models.py:183]: Am I right that there is no any test to check deprecated param?

**Review by idegtiarov — COMMENTED** [review]
One question, other LGTM!

File: common/djangoapps/student/models.py (hunk 1 above)
Comment (dyudyunov) [reply common/djangoapps/student/models.py:183]: I've checked tests using the search, run the student module tests locally and hope to find out if any test will fail using the GH pipeline

Comment (dyudyunov) [reply common/djangoapps/student/models.py:183]: waiting for the pipeline to finish to ensure all  is ok

Comment (idegtiarov) [reply common/djangoapps/student/models.py:183]: Great, then waiting until tests passed and open PR to upstream

**Review by idegtiarov — APPROVED** [review]

**Pull request closed without merging**